recipes/gcc/all/conanfile.py

```diff
@@ -0,0 +1,131 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanException, ConanInvalidConfiguration
+import os
+
+
+class GccConan(ConanFile):
+    name = "gcc"
+    description = "The GNU Compiler Collection includes front ends for C, " \
+                  "C++, Objective-C, Fortran, Ada, Go, and D, as well as " \
+                  "libraries for these languages (libstdc++,...). "
+    topics = ("gcc", "gnu", "compiler", "c", "c++")
+    homepage = "https://gcc.gnu.org"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "GPL-3.0-only"
+    settings = "os", "compiler", "arch", "build_type"
+    _autotools = None
+
+    def build_requirements(self):
+        self.build_requires("flex/2.6.4")
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        pkgversion = 'conan GCC %s' % self.version
+        bugurl = self.url + '/issues'
+        libdir = "%s/lib/gcc/%s" % (self.package_folder, self.version)
+        args = [
+            "--enable-languages=c,c++",
+            "--disable-nls",
+            "--disable-multilib",
+            "--disable-bootstrap",
+            "--with-system-zlib",
+            "--with-gmp=%s" % self.deps_cpp_info['gmp'].rootpath,
+            '--with-mpc=%s' % self.deps_cpp_info["mpc"].rootpath,
+            "--with-mpfr=%s" % self.deps_cpp_info["mpfr"].rootpath,
+            "--without-isl",
+            "--libdir=%s" % libdir,
+            '--with-pkgversion=%s' % pkgversion,
+            "--program-suffix=-%s" % self.version,
+            "--with-bugurl=%s" % bugurl
+        ]
+        if self.settings.os == "Macos":
+            xcrun = tools.XCRun(self.settings)
+            args.extend([
+                '--with-native-system-header-dir=/usr/include',
+                "--with-sysroot={}".format(xcrun.sdk_path)
+            ])
+        self._autotools.libs = []  # otherwise causes config.log to fail finding -lmpc
+        if self.settings.compiler in ["clang", "apple-clang"]:
+            # xgcc: error: unrecognized command-line option -stdlib=libc++
+            if self.settings.compiler.libcxx == "libc++":
+                self._autotools.cxx_flags.remove("-stdlib=libc++")
+            elif self.settings.compiler.libcxx in ["libstdc++", "libstdc++11"]:
+                self._autotools.cxx_flags.remove("-stdlib=libstdc++")
+        self._autotools.configure(args=args, configure_dir=self._source_subfolder)
+        return self._autotools
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def requirements(self):
+        self.requires("mpc/1.1.0")
+        self.requires('mpfr/4.0.2')
```

Comment:
```suggestion
        self.requires('mpfr/4.1.0')
```

WARN: mpc/1.1.0: requirement mpfr/4.1.0 overridden by gcc/10.0.1 to mpfr/4.0.2